Invoke event error log

FuzzB0t opened the pull request.

dbeal-eth reviewed (changes requested):
It's not necessary to admit a warning when events are admitted, but no event is expected. That is very normal and should not be anything that we need to admit a warning for.

FuzzB0t commented:
Reference Issue #268

FuzzB0t commented:
Issue #268

Before when running the build cmd for a cannonfile that contained an invoke step which expected an event from an invoked function, cannon would execute the step without any warning:

![image](https://github.com/usecannon/cannon/assets/30664234/3adb011c-3270-478e-a287-b84e3aca0901)

The above consumer cannonfile expects an event, meanwhile the contract function thats being invoked is not emitting an event, After rebuilding the contract and running build on the consumer cannonfile the output would look like this:

![image](https://github.com/usecannon/cannon/assets/30664234/204aa8b1-8d68-4c38-9b8d-1b3d923f80c4)

FuzzB0t commented:
After making changes to the invoke step, the build cmd will now throw an error when an event is expected but not emitted. You can choose to ignore these errors by adding the 'ignoreErrors' flag under an invoke step:
![image](https://github.com/usecannon/cannon/assets/30664234/f6b3f948-bbcf-4280-8525-a107ebccd4aa)


So by running the build cmd on the context above we now get:

![image](https://github.com/usecannon/cannon/assets/30664234/49f3c045-76a3-4e61-aac8-fe508ae05069)

FuzzB0t commented:
Issue #268 

For reference, here's the same test run on the hardhat sample project to test out the changes on the hardhat-plugin:

![image](https://github.com/usecannon/cannon/assets/30664234/adf538f2-6ac4-4b66-bd48-6bafc6b7a52c)

![image](https://github.com/usecannon/cannon/assets/30664234/c46bb308-828b-437a-8bc5-abdb24fe1ac1)

FuzzB0t commented:
Added an example of multiple event emissions in the Greeter contract under the sample foundry contract:

Multiple events get emitted from one TX and the cannonfile uses an event from a previous invoke step to set attributes for a subsequent invoke step:

> Greeter Contracts

![image](https://github.com/usecannon/cannon/assets/30664234/68cee413-2506-4a4f-b3aa-a31cd9c75020)

> Cannonfile.consumer.toml

![image](https://github.com/usecannon/cannon/assets/30664234/408be849-c59a-4427-9e80-62a566626667)

> Build command output log

![image](https://github.com/usecannon/cannon/assets/30664234/bb553eec-2e28-41fb-97a0-bf0c2c5bf506)

dbeal-eth reviewed (approved):
Awesome

FuzzB0t commented:
Added a few more examples to simply show the multiple events / multiple contract deployments referencing within a cannonfile:

![image](https://github.com/usecannon/cannon/assets/30664234/5c2f08dc-8f52-4044-92e4-aae17e7a4c2f)

![image](https://github.com/usecannon/cannon/assets/30664234/5efa2427-08d0-4cd0-bfd4-7b5429563b64)

Also updated the technical docs with info about the extra option in the cannonfile and how to reference events and contracts in subsequent invoke steps.
